sql/atxs/atxs_test.go

```diff
@@ -981,3 +981,56 @@ func TestLatest(t *testing.T) {
 		})
 	}
 }
+
+func Test_PrevATXCollisions(t *testing.T) {
+	db := sql.InMemory()
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+
+	// create two ATXs with the same PrevATXID
+	prevATXID := types.RandomATXID()
+
+	atx1 := newAtx(t, sig, withPublishEpoch(1), withPrevATXID(prevATXID))
+	atx2 := newAtx(t, sig, withPublishEpoch(2), withPrevATXID(prevATXID))
+
+	require.NoError(t, atxs.Add(db, atx1))
+	require.NoError(t, atxs.Add(db, atx2))
+
+	// verify that the ATXs were added
+	got1, err := atxs.Get(db, atx1.ID())
+	require.NoError(t, err)
+	atx1.AtxBlob = types.AtxBlob{}
+	require.Equal(t, atx1, got1)
+
+	got2, err := atxs.Get(db, atx2.ID())
+	require.NoError(t, err)
+	atx2.AtxBlob = types.AtxBlob{}
+	require.Equal(t, atx2, got2)
+
+	// add 10 valid ATXs by 10 other smeshers
+	atxMap := make(map[types.NodeID][]*types.ActivationTx)
+	for i := 2; i < 12; i++ {
+		otherSig, err := signing.NewEdSigner()
+		require.NoError(t, err)
+
+		if len(atxMap[otherSig.NodeID()]) == 0 {
+			atx := newAtx(t, otherSig, withPublishEpoch(types.EpochID(i)))
+			require.NoError(t, atxs.Add(db, atx))
+		} else {
+			atx := newAtx(t, otherSig,
+				withPublishEpoch(types.EpochID(i)),
+				withPrevATXID(atxMap[otherSig.NodeID()][len(atxMap[otherSig.NodeID()])-1].ID()),
+			)
+			require.NoError(t, atxs.Add(db, atx))
+		}
+	}
```

Comment:
The `atxMap` is always empty. Is the code meant to handle the rare case when `signing.NewEdSigner()` creates the same identity for the second time? I think that's nearly not possible as the keys are seeded from rand. How about simplifying to:

```suggestion
	for i := 2; i < 12; i++ {
		otherSig, err := signing.NewEdSigner()
		require.NoError(t, err)
		atx := newAtx(t, otherSig, withPublishEpoch(types.EpochID(i)))
		require.NoError(t, atxs.Add(db, atx))
	}
```

Reply:
The idea was to have multiple ATXs by the same NodeIDs. Don't know what went wrong here - I updated the test accordingly.